src/org.hdfgroup.object/hdf/object/h5/H5ScalarDS.java

```diff
@@ -908,7 +904,21 @@ private Object scalarDatasetCommonIO(H5File.IO_TYPE ioType, Object writeBuf) thr
 
                 if (ioType == H5File.IO_TYPE.READ) {
                     log.trace("scalarDatasetCommonIO():read ioType isNamed={} isEnum={} isText={} isRefObj={}", dsDatatype.isNamed(), dsDatatype.isEnum(), dsDatatype.isText(), dsDatatype.isRefObj());
-                    if ((originalBuf == null) || dsDatatype.isEnum() || dsDatatype.isText() || dsDatatype.isRefObj()
+                    if (dsDatatype.isVarStr()) {
+                        try {
+                            theData = H5Datatype.allocateArray(dsDatatype, (int)totalSelectedSpacePoints);
```

Comment:
This just does the same thing as the code below for enums, "text" datasets, object references, etc.

Reply:
Oh, yeah, I needed to check VL type first before the others I think. OR maybe I needed to check VarStr before VLEN? 
I'll double check if there is a dependency issue.

Reply:
As much as I'm not a fan of how big these if checks are getting (it should be abstractable through classes or other methods), I think `if (dsDatatype.isVLEN() && !dsDatatype.isVarStr())` should be fine for the VL case.

Reply:
Yes I think I did that elsewhere.